internal/wire/max_data_frame.go

```diff
@@ -0,0 +1,51 @@
+package wire
+
+import (
+	"bytes"
+
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+	"github.com/lucas-clemente/quic-go/internal/utils"
+)
+
+// A MaxDataFrame carries flow control information for the connection
+type MaxDataFrame struct {
+	ByteOffset protocol.ByteCount
+}
+
+// ParseMaxDataFrame parses a MAX_DATA frame
+func ParseMaxDataFrame(r *bytes.Reader, version protocol.VersionNumber) (*MaxDataFrame, error) {
+	// read the TypeByte
+	if _, err := r.ReadByte(); err != nil {
+		return nil, err
+	}
+
+	frame := &MaxDataFrame{}
+	byteOffset, err := utils.GetByteOrder(version).ReadUint64(r)
+	if err != nil {
+		return nil, err
+	}
+	frame.ByteOffset = protocol.ByteCount(byteOffset)
+	return frame, nil
+}
+
+//Write writes a MAX_STREAM_DATA frame
+func (f *MaxDataFrame) Write(b *bytes.Buffer, version protocol.VersionNumber) error {
+	if version.CryptoStreamID() != 0 {
```

Comment:
Are we sure that MAX_DATA will appear in the same version as stream 0 for the crypto stream?

If yes, then I'd still add a bool-valued function to the version, so that we don't need this comparison everywhere.

Reply:
I think there's no other way. How else would you distinguish a WINDOW_UPDATE for stream 0 and the connection?